### ISSUE TITLE
Service hold-off time over for key service

### ISSUE THREAD

**Issue by nh2** [NixOS/nixops#844]
Today a deploy of mine failed, and I noticed this in the logs:

```
# journalctl -u gluster-ssl-privkey-key.service
-- Logs begin at Sun 2018-01-21 20:05:49 UTC, end at Sun 2018-01-21 21:29:06 UTC. --
Jan 21 20:11:55 node-1 systemd[1]: Starting gluster-ssl-privkey-key.service...
Jan 21 20:11:55 node-1 gluster-ssl-privkey-key-pre-start[1162]: flapped down
Jan 21 20:11:55 node-1 systemd[1]: Started gluster-ssl-privkey-key.service.
Jan 21 20:14:29 node-1 systemd[1]: gluster-ssl-privkey-key.service: Service hold-off time over, scheduling restart.
Jan 21 20:14:29 node-1 systemd[1]: Stopped gluster-ssl-privkey-key.service.
Jan 21 20:14:29 node-1 systemd[1]: Starting gluster-ssl-privkey-key.service...
Jan 21 20:14:29 node-1 gluster-ssl-privkey-key-pre-start[3776]: flapped down
Jan 21 20:14:29 node-1 systemd[1]: Started gluster-ssl-privkey-key.service.
```

Why would `Service hold-off time over, scheduling restart` be emitted, given that it has `Started` successfully?

I don't understand which systemd timer this is. The [key service](https://github.com/NixOS/nixops/blob/379b1f933b930aafc8d386d8bfec97ce0e775628/nix/keys.nix#L246-L270) doesn't seem to have anything that would make this occur.

Any ideas?

**Comment by nh2** [comment]
Another thing that's probably independent of this issue but I also don't understand it:

In

https://github.com/NixOS/nixops/blob/379b1f933b930aafc8d386d8bfec97ce0e775628/nix/keys.nix#L255-L259

it uses `kill %1` to terminate the `inotify-wait` before calling `exit`, but just below in

https://github.com/NixOS/nixops/blob/379b1f933b930aafc8d386d8bfec97ce0e775628/nix/keys.nix#L265-L268

there is no `kill %1` to do the same. Why is that?

**Comment by nh2** [comment]
CC @ryanartecona for the question above, who added that code in https://github.com/NixOS/nixops/commit/5308d265438249df6ef51e49a8d8d9fb8833471c.

**Comment by nh2** [comment]
Ah, the hold-off timer happens [when the service exits](https://access.redhat.com/discussions/2944541). I guess this is what happens in my log from the issue description:

https://github.com/NixOS/nixops/blob/379b1f933b930aafc8d386d8bfec97ce0e775628/nix/keys.nix#L263-L269

The nixops side [deletes the key](https://github.com/NixOS/nixops/blob/379b1f933b930aafc8d386d8bfec97ce0e775628/nixops/backends/__init__.py#L236), and thus `inotifywait -e delete_self` terminates, and thus `wait %1` terminates, without printing anything further, and [100 ms later](https://github.com/NixOS/nixops/blob/379b1f933b930aafc8d386d8bfec97ce0e775628/nix/keys.nix#L249) systemd prints the hold-off timer message.

I guess we should `echo` something after the `wait %1` , e.g. `key got deleted` and for the other case `key got created`, to make reading what happens easier.

I'll send a PR.

**Comment by ryanartecona** [comment]
@nh2 Yeah, that matches my vague memory of what that code does. I think those extra log lines would be a good improvement.
